### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by RobLoach**
This updates the dependencies, and switches to [Composer scripts](https://getcomposer.org/doc/articles/scripts.md) for running phpunit.

Still resulting in a failure, not quite sure why. Any ideas?

### CODE REVIEW

File: tests/PluginTest.php [hunk 1]
```diff
@@ -128,13 +128,12 @@ protected function composerJSONDefaults() {
...
         'drupal-composer/drupal-scaffold' => $this->tmpReleaseTag,
-        'composer/installers' => '^1.0.20',
-        'drupal/core' => '8.0.0',
+        'composer/installers' => '^1.0.24',
+        'drupal/core' => '8.1.0-rc1',
       ),
       'scripts' => array(
         'drupal-scaffold' =>  'DrupalComposer\\DrupalScaffold\\Plugin::scaffold'
-      ),
-      'minimum-stability' => 'dev',
```
Comment (webflo) [inline tests/PluginTest.php]: The minimum-stability flag is required. The test can't resolve ``'drupal/core' => '8.1.0-rc1'`` otherwise.

Comment (RobLoach) [inline tests/PluginTest.php]: Testing with `8.1.*@rc` rather then minimum stability.

**Comment by webflo** [comment]
@RobLoach Please rebase against the master branch.

**Comment by bradjones1** [comment]
While we're at it, the constraint on robo appears too strict, when paired with the master branch of Drush (9.x) - they are requiring `~1.0.0-beta1`, which can't resolve with `^0.7.1`.

Perhaps `>0.7.1` would be appropriate?

**Comment by webflo** [comment]
@greg-1-anderson Can we update to robo 1?

// cc @phenaproxima

**Comment by phenaproxima** [comment]
acquia/lightning, which requires both Drush and drupal-scaffold, is being blocked by the fact that both projects require Robo, but in incompatible versions. This is a critical blocker for us; it may even be blocking people from installing Lightning, let alone us developing it further.

**Comment by greg-1-anderson** [comment]
@webflow: I think we should wait for Robo 1.0.0 stable; that should be available in a week or so.

@phenaproxima:  Do you need to use Drush dev-master?  If you use a stable release of Drush, then you should not have a problem.

We'll make sure that both projects are using Robo ~1 once its stable release is available.

**Comment by phenaproxima** [comment]
@greg-1-anderson: We do need dev-master, because for development purposes we depend upon the composer.lock compatibility of the make-convert command. Which, to our knowledge, is only available in dev-master.

**Comment by greg-1-anderson** [comment]
@phenaproxima: I did not confirm, but did you try drush/drush: 9.0.0-alpha1? That release should be very similar to dev-master, but without the Robo dependency, IIRC.

**Comment by phenaproxima** [comment]
@greg-1-anderson: I'll give it a shot!

**Comment by phenaproxima** [comment]
@greg-1-anderson: I have tried it and, unfortunately, it doesn't work.

At some point (a month or two ago?) @grasmash got a PR merged (drush-ops/drush#1933) into Drush which allows the make-convert command to support composer.lock files. It doesn't look like Drush 9.0.0-alpha1 includes this commit, which Lightning unfortunately depends on for development. That's why we're still depending on dev-master.

I'm going to try to have Composer patch Drush to get the command we need.

**Comment by bradjones1** [comment]
I'm sure it's not elegant nor preferred, but in the meantime requiring `"codegyre/robo": "1.0.0-beta1 as 0.7.2"` appears to make composer happy with drush at `dev-master` and drupal-scaffold at `^1.3.1`.

**Comment by webflo** [comment]
Addressed in #38

**Pull request closed without merging**